### PR TITLE
Fix UTF issues in 2019.2

### LINKED ISSUE #44: Travis CI - snap state fails
Travis CI has following failed state for Ubuntu:

```
      ----------
                 ID: packages-snapd-hello-world-wanted
           Function: cmd.run
               Name: snap install hello-world
             Result: False
            Comment: Command "snap install hello-world" run
            Started: 14:24:07.177879
           Duration: 43.539 ms
            Changes:   
              ----------
              pid:
                  7842
              retcode:
                  10
              stderr:
                  /bin/bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)
                  error: too early for operation, device not yet seeded or device model not acknowledged
              stdout:
```

### LINKED ISSUE #48: problem with UTF?
I'm using this formula and since I installed salt 2019.2.0 I have this problem (tested in ubuntu 16.04 and redhat 6):
```
# salt-call state.apply packages 
[INFO    ] Loading fresh modules for state activity
[INFO    ] Running state [pkg_req_pkgs] at time 15:29:33.789483
[INFO    ] Executing state pkg.installed for [pkg_req_pkgs]
[INFO    ] Executing command [u'dpkg-query', u'--showformat', u'${Status} ${Package} ${Version} ${Architecture}', u'-W'] in directory '/root'
[INFO    ] Executing command [u'apt-cache', u'-q', u'policy', u"u'curl'"] in directory '/root'
[INFO    ] Executing command [u'apt-cache', u'-q', u'policy', u"u'gzip'"] in directory '/root'
[INFO    ] Executing command [u'apt-cache', u'-q', u'policy', u"u'bzip2'"] in directory '/root'
[INFO    ] Executing command [u'apt-get', u'-q', u'update'] in directory '/root'
[INFO    ] Executing command [u'dpkg', u'--get-selections', u'*'] in directory '/root'
[INFO    ] Executing command [u'systemd-run', u'--scope', u'apt-get', u'-q', u'-y', u'-o', u'DPkg::Options::=--force-confold', u'-o', u'DPkg::Options::=--force-confdef', u'install', u"u'curl'", u"u'gzip'", u"u'bzip2'"] in directory '/root'
[ERROR   ] Command '[u'systemd-run', u'--scope', u'apt-get', u'-q', u'-y', u'-o', u'DPkg::Options::=--force-confold', u'-o', u'DPkg::Options::=--force-confdef', u'install', u"u'curl'", u"u'gzip'", u"u'bzip2'"]' failed with return code: 100
[ERROR   ] stdout: Reading package lists...
Building dependency tree...
Reading state information...
[ERROR   ] stderr: Running scope as unit run-r08177169660649899531c302403fa65e.scope.
E: Unable to locate package u'curl'
E: Unable to locate package u'gzip'
E: Unable to locate package u'bzip2'
[ERROR   ] retcode: 100
[INFO    ] Executing command [u'dpkg-query', u'--showformat', u'${Status} ${Package} ${Version} ${Architecture}', u'-W'] in directory '/root'
[ERROR   ] Problem encountered installing package(s). Additional info follows:

errors:
    - Running scope as unit run-r08177169660649899531c302403fa65e.scope.
      E: Unable to locate package u'curl'
      E: Unable to locate package u'gzip'
      E: Unable to locate package u'bzip2'
```

I think there is a problem with UTF and python because the formula try to install the package  u'curl' instead of curl.

Thank you

EDIT: same problem with ubuntu 18.04 with salt python2, but there is no problem with ubuntu 18.04 and salt with python3

### PR DESCRIPTION
**Pull request by javierbertoli**
This PR should fix #48.
Added new distros to test, following [template-formula](https://github.com/saltstack-formulas/template-formula)
Fix/disabled tests where failing (particularly #44)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -10,8 +10,9 @@ before_install:
 
 env:
   matrix:
-    - INSTANCE: deb-debian-9
-    - INSTANCE: deb-ubuntu-1804
+    - INSTANCE: debian-debian-9
+    - INSTANCE: ubuntu-ubuntu-1604
+    - INSTANCE: ubuntu-ubuntu-1804
     - INSTANCE: centos-centos-7
```
Comment (myii) [inline .travis.yml]: How do you feel about adding all of the rest of the platforms that we've got working on `template-formula`?

**Comment by myii** [comment]
@javierbertoli Eyeball check: looks good, thanks.

File: kitchen.yml [hunk 2]
```diff
@@ -57,14 +62,26 @@ verifier:
...
+      - ubuntu-18.04
+
+  - name: ubuntu
+    excludes:
+      - debian-9
+      - fedora-27
+      - centos-7
+    provisioner:
+      pillars-from-files:
+        packages.sls: test/integration/default/pillar.example.ubuntu
```
Comment (myii) [inline kitchen.yml]: This is causing the tests to fail for `16.04` and `18.04`:

```travis
Failed to complete #converge action: [No such file or directory @ rb_sysopen - test/integration/default/pillar.example.ubuntu] on ubuntu-ubuntu-1604
```

File: kitchen.yml [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+provisioner:
+  name: salt_solo
+  log_level: info
+  require_chef: false
+  salt_version: latest
+  formula: packages
+  salt_copy_filter:
+    - .kitchen
+    - .git
+  pillars-from-files:
```
Comment (myii) [inline kitchen.yml]: Also noticed in the logs:

* `pillars-from-files` is deprecated in favor of `pillars_from_files`

File: kitchen.yml [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+
+  - name: ubu16
+    excludes:
+      - debian-9
+      - ubuntu-18.04
+      - centos-7
+      - fedora
+      - opensuse-leap-salt-minion
+    provisioner:
+      pillars-from-files:
```
Comment (myii) [inline kitchen.yml]: `pillars_from_files`

File: kitchen.yml [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+
+  - name: fed
+    excludes:
+      - debian-9
+      - ubuntu-16.04
+      - ubuntu-18.04
+      - centos-7
+      - opensuse-leap-salt-minion
+    provisioner:
+      pillars-from-files:
```
Comment (myii) [inline kitchen.yml]: `pillars_from_files`

File: kitchen.yml [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+      dependencies:
+        - name: epel
+          repo: git
+          source: https://github.com/saltstack-formulas/epel-formula.git
+      state_top:
+        base:
+          '*':
+            - epel
+            - packages
+      pillars-from-files:
```
Comment (myii) [inline kitchen.yml]: `pillars_from_files`

File: .travis.yml (hunk 1 above)
Comment (javierbertoli) [reply .travis.yml]: Added them.

File: kitchen.yml (hunk 2 above)
Comment (javierbertoli) [reply kitchen.yml]: Typo in the naming. Fixed.

File: kitchen.yml (hunk 3 above)
Comment (javierbertoli) [reply kitchen.yml]: Fixed

**Comment by myii** [comment]
@javierbertoli Appreciate the effort you've put into this, everything looking good.  Merged.

**Comment by javierbertoli** [comment]
Thanks, @myii

**Pull request merged**